test/front_desk_test.rb

```diff
@@ -0,0 +1,117 @@
+require_relative "test_helper"
+# to do: refactor for more clarity - ex - @fd & @front_desk
+describe Hotel::FrontDesk do
+  before do
+    @front_desk = Hotel::FrontDesk.new
+    @date = Date.parse("2020-08-04")
+  end
+
+  describe "rooms" do
+    it "returns a list" do
+      rooms = @front_desk.rooms
+      expect(rooms).must_be_kind_of Array
+    end
+  end
+  
+  describe "reserve_room" do
+    it "takes two Date objects and adds reservation to room" do
+      start_date = @date
+      end_date = start_date + 3
+      reservation = @front_desk.reserve_room(start_date, end_date)
+      expect(reservation).must_be_kind_of Hotel::Reservation
+      room = @front_desk.rooms.find {|r| r.reservations.length > 0}
+      expect(room.reservations.include? reservation).must_equal true
+    end
+
+    it "successfully reserves a room when all rooms have at least one reservation" do
+      (1..20).each do
+        @front_desk.reserve_room(Date.new(2020, 01,01), Date.new(2020,01,03))
+      end
+      reservation21 = @front_desk.reserve_room(Date.new(2020, 01,03), Date.new(2020,01,04))
+      room_with_21 = @front_desk.rooms.find {|room| room.reservations.length == 2}
+      expect(room_with_21.reservations[1]).must_equal reservation21
+    end
+
+    it "raises argument error when no rooms are available" do
+      (1..20).each do
+        @front_desk.reserve_room(Date.new(2020, 01,01), Date.new(2020,01,03))
+      end
+      expect{@front_desk.reserve_room(Date.new(2020, 01,01), Date.new(2020,01,03))}.must_raise ArgumentError
+    end
+
```

Comment:
Your `block` tests are quite comprehensive. Now that you have implemented blocks, you should also add a test here to make sure you can't reserve a room that's part of a block.